### ISSUE TITLE
Duplicate anchors when type variables exist in type signatures

### ISSUE THREAD

**Issue by jinwoo** [google/haskell-indexer#123]
E.g., in
```haskell
module TypeVarInSig where

isJust :: Maybe a -> Bool                                                                
isJust (Just x) = True                                                                   
isJust Nothing = False                                                                   
```
, the type variable `a` in the type signature of `isJust` generates two identical anchor nodes. And it causes the kythe-verifier to fail. The error looks like this:

```
Verifying: testdata/basic/TypeVarInSig.hs
Two facts were equal:
  App(fact, (App(vname, (haskell:type:main_main:TypeVarInSig:a:testdata/basic/TypeVarInSig.hs:(3,1)-(3,26):43,44, "", "", testdata/basic/TypeVarInSig.hs, haskell)), "", "", /kythe/node/kind, anchor))
  App(fact, (App(vname, (haskell:type:main_main:TypeVarInSig:a:testdata/basic/TypeVarInSig.hs:(3,1)-(3,26):43,44, "", "", testdata/basic/TypeVarInSig.hs, haskell)), "", "", /kythe/node/kind, anchor))
Two facts were equal:
  App(fact, (App(vname, (haskell:type:main_main:TypeVarInSig:a:testdata/basic/TypeVarInSig.hs:(3,1)-(3,26):43,44, "", "", testdata/basic/TypeVarInSig.hs, haskell)), "", "", /kythe/loc/start, 43))
  App(fact, (App(vname, (haskell:type:main_main:TypeVarInSig:a:testdata/basic/TypeVarInSig.hs:(3,1)-(3,26):43,44, "", "", testdata/basic/TypeVarInSig.hs, haskell)), "", "", /kythe/loc/start, 43))
Two facts were equal:
  App(fact, (App(vname, (haskell:type:main_main:TypeVarInSig:a:testdata/basic/TypeVarInSig.hs:(3,1)-(3,26):43,44, "", "", testdata/basic/TypeVarInSig.hs, haskell)), "", "", /kythe/loc/end, 44))
  App(fact, (App(vname, (haskell:type:main_main:TypeVarInSig:a:testdata/basic/TypeVarInSig.hs:(3,1)-(3,26):43,44, "", "", testdata/basic/TypeVarInSig.hs, haskell)), "", "", /kythe/loc/end, 44))
```

**Comment by jinwoo** [comment]
It looks like one of the anchors is for the declaration and the other is for the xref. @robinp , is there a reason to have a xref here?

**Comment by jinwoo** [comment]
To make it clearer, the xref points to itself.

**Comment by jinwoo** [comment]
Noting what I found so far. The decl in this case is an "implicit" decl because type vars don't have explicit definition sites. So it's correct to have both the decl node and the xref node. The problem is that currently both have the same span.

I saw this comment: https://github.com/google/haskell-indexer/blob/master/haskell-indexer-backend-ghc/src/Language/Haskell/Indexer/Backend/Ghc.hs#L254

As noted there, we can make those implicit decls have no spans. When I do that, this problem goes away.
